test_runner/src/main/kotlin/ftl/doctor/Doctor.kt

```diff
@@ -61,3 +67,10 @@ private fun preloadConfiguration(data: Path, isAndroid: Boolean) =
     } catch (e: FlankConfigurationError) {
         e.message
     }
+
+private fun JsonNode.validateDevices() = StringBuilder().apply {
+    getDevicesNode()?.getNotValidDevices()?.forEach { device ->
+        if (device.has(VERSION_NODE))
+            appendLine("Warning: Version should be string $GCLOUD_NODE -> $DEVICES_NODE[${device[MODEL_NODE]}] -> $VERSION_NODE[${device[VERSION_NODE]}]")
+    }
```

Comment:
How about filter and skip `if` ?

```suggestion
getDevicesNode()?.getNotValidDevices()?.filter {device.has(VERSION_NODE)}?.forEach { device ->
            appendLine("Warning: Version should be string $GCLOUD_NODE -> $DEVICES_NODE[${device[MODEL_NODE]}] -> $VERSION_NODE[${device[VERSION_NODE]}]")
    }
```

Reply:
👍